src/sentry/integrations/slack/views/unlink_team.py

```diff
@@ -51,32 +56,59 @@ class SlackUnlinkTeamView(BaseView):
     Django view for unlinking team from slack channel. Deletes from ExternalActor table.
     """
 
+    _METRICS_SUCCESS_KEY = "sentry.integrations.slack.unlink_team_view.success"
+    _METRICS_FAILURE_KEY = "sentry.integrations.slack.unlink_team_view.failure"
+
     @method_decorator(never_cache)
-    def handle(self, request: Request, signed_params: str) -> HttpResponse:
+    def dispatch(self, request: HttpRequest, signed_params: str) -> HttpResponse:
         if request.method not in ALLOWED_METHODS:
             return HttpResponse(status=405)
 
         try:
-            params = unsign(signed_params)
-        except (SignatureExpired, BadSignature):
+            converted = unsign(signed_params)
+            params = TeamIdentityParams(**converted)
+        except (SignatureExpired, BadSignature) as e:
+            _logger.warning("dispatch.signature_error", exc_info=e)
+            metrics.incr(self._METRICS_FAILURE_KEY, tags={"error": str(e)}, sample_rate=1.0)
             return render_to_response(
                 "sentry/integrations/slack/expired-link.html",
                 request=request,
             )
-
-        integration = integration_service.get_integration(integration_id=params["integration_id"])
+        except TypeError:
+            _logger.exception("dispatch.type_error")
+            metrics.incr(self._METRICS_FAILURE_KEY, sample_rate=1.0)
+            return render_error_page(request, status=400, body_text="HTTP 400: Invalid parameters")
+
+        logger_params = {
+            "organization_id": params.organization_id,
+            "integration_id": params.integration_id,
+            "channel_id": params.channel_id,
+            "channel_name": params.channel_name,
+            "slack_id": params.slack_id,
```

Comment:
you could do something like `asdict(params)` here instead of typing all of the fields in params out!